src/python/pants/util/objects.py

```diff
@@ -276,20 +275,30 @@ def _allowed_values(cls):
       return list(cls._singletons.keys())
 
     def __new__(cls, value):
-      """Forward `value` to the .create() factory method.
+      """Create an instance of this enum.
 
-      The .create() factory method is preferred, but forwarding the constructor like this allows us
-      to use the generated enum type both as a type to check against with isinstance() as well as a
-      function to create instances with. This makes it easy to use as a pants option type.
+      :param value: Use this as the enum value. If `value` is an instance of this class, return it,
+                    otherwise it is checked against the enum's allowed values.
       """
-      return cls.create(value)
+      if isinstance(value, cls):
```

Comment:
Why do we allow this? Are there places where we don't know if the caller has a string of an enum value? If so, can we just fix them, and only accept strings here?

Reply:
This is a good point. I will remove this conditional and see what breaks. I think it might end up being nothing.